src/components/UserSelector.tsx

```diff
@@ -1,35 +1,91 @@
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
+import { User } from '../types/User';
+
+type Props = {
+  users: User[];
+  select: User | null;
+  onSelect: React.Dispatch<React.SetStateAction<User | null>>
+  getUserPosts: (userId: number) => void;
+  setIsCommentListHidden: React.Dispatch<React.SetStateAction<boolean>>;
+};
+
+export const UserSelector: React.FC<Props> = ({
+  users,
+  select,
+  onSelect,
+  getUserPosts,
+  setIsCommentListHidden,
+}) => {
+  const [isVisible, setIsVisible] = useState(false);
+  const selectRef = useRef(null);
+
+  const handleSelectUser = (user: User) => {
+    onSelect(user);
```

Comment:
Avoid invoking multiple state change functions sequentially as it may cause unnecessary re-renders. Consider using a reducer to manage related states.